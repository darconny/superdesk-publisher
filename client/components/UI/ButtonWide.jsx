import React from 'react'
import PropTypes from 'prop-types'

const ButtonWide = ({ onClick, label }) => {
    return <button className="btn btn--hollow btn--expanded btn--primary" onClick={onClick}>{label}</button>
}

ButtonWide.propTypes = {
    onClick: PropTypes.func,
    label: PropTypes.string
}

ButtonWide.defaultProps = {
    label: 'Add',
}

export default ButtonWide