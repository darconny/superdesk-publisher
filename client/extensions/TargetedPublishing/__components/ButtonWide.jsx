import React from 'react'
import PropTypes from 'prop-types'

function ButtonWide({ onClick, label }) {
    return <button className="btn btn--hollow btn--expanded btn--primary" onClick={onClick}>{label}</button>
}

ButtonWide.propTypes = {
    onClick: PropTypes.func,
    label: PropTypes.string
}

ButtonWide.defaultProps = {
    label: 'Add',
}

export default ButtonWide