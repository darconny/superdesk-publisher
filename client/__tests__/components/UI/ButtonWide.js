import React from 'react'
import ButtonWide from '../../../components/UI/ButtonWide'
import { render, fireEvent } from '@testing-library/react'

it('UI/ButtonWide renders correctly', () => {
   const {container} = render(<ButtonWide label="Label"/>)

   expect(container.firstChild).toMatchSnapshot()
})

it('UI/ButtonWide onClick function fired', () => {
    const onClick = jest.fn()
    const { container } = render(<ButtonWide onClick={onClick} />)
    const button = container.querySelector('button')

    fireEvent.click(button)
    expect(onClick).toHaveBeenCalled()
 })
