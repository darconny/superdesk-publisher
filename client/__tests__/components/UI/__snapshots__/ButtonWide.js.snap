// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI/ButtonWide renders correctly 1`] = `
<button
  class="btn btn--hollow btn--expanded btn--primary"
>
  Label
</button>
`;
